src/common/leaky_bucket.cpp

```diff
@@ -43,15 +45,21 @@ double LeakyBucket::Bucket::Add(double tokens) {
     if (avg == 0) {
         return 0;
     }
-
+    VLOG(9) << ": avg is: " << avg
+            << ", level is: " << level
+            << ", burst is: " << burst
+            << ", burstLevel is: " << burstLevel
+            << ", token is: " << tokens;
     double available = 0;
-
+    levelInitial = false;
+    burstLevelInitial = false;
     if (burst > 0) {
         // if burst is enabled, available tokens is limit by two conditions
         // 1. bucket-level is limited by bucket capacity which is calculate by [burst * burstLength]  // NOLINT
         // 2. bucket-burst-level is limited by burst limit which is equal to [burst]  // NOLINT
+        double capacity = burst + (burst - avg) * (burstLength - 1);
```

Comment:
capacity can be defined as a member variable that is copied at initialization rather than evaluated each time an Add is called.

Reply:
fix